### PR TITLE
add extra-data encoder/decoder tool

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: cmd/istanbul/cmd.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	atypes "github.com/getamis/go-ethereum/core/types"
+	"github.com/naoina/toml"
+	"github.com/urfave/cli"
+)
+
+var (
+	decodeCommand = cli.Command{
+		Action:    decode,
+		Name:      "decode",
+		Usage:     "To decode a Istanbul extraData",
```
Comment (markya0616) [inline cmd/istanbul/cmd.go]: decode an

File: README.md [hunk 2]
```diff
@@ -1 +1,66 @@
...
+COMMANDS:
+     decode   To decode a Istanbul extraData
+     encode   To encode a Istanbul extraData
+     help, h  Shows a list of commands or help for one command
+
+GLOBAL OPTIONS:
+   --help, -h  show help
+
+COPYRIGHT:
+   Copyright 2017 The Getamis Authors
```
Comment (yutelin) [inline README.md]: rename Getamis to Amis

File: cmd/istanbul/main.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Istanbul-tools/cmd/utils"
```
Comment (alanchchen) [inline cmd/istanbul/main.go]: "github.com/getamis/Istanbul-tools/cmd/utils" ?

File: cmd/istanbul/cmd.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		ArgsUsage: "<extra data>",
+		Flags: []cli.Flag{
+			ExtraDataFlag,
+		},
+		Description: `
+The extraData command will decode extraData for the given input which should be a hex string.
```
Comment (markya0616) [inline cmd/istanbul/cmd.go]: This command decodes extraData to vanity and validators

File: cmd/istanbul/cmd.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	encodeCommand = cli.Command{
+		Action:    encode,
+		Name:      "encode",
+		Usage:     "To encode a Istanbul extraData",
+		ArgsUsage: "<config file>",
+		Flags: []cli.Flag{
+			ConfigFlag,
+		},
+		Description: `
+The extraData command will encode Istanbul extraData for the given input file.
```
Comment (markya0616) [inline cmd/istanbul/cmd.go]: This command encodes vanity and validators to extraData. Please refer to example/config.toml.

File: cmd/istanbul/cmd.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rlp"
+	atypes "github.com/getamis/go-ethereum/core/types"
```
Comment (markya0616) [inline cmd/istanbul/cmd.go]: Why rename to atypes?

File: cmd/istanbul/flags.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package main
```
Comment (markya0616) [inline cmd/istanbul/flags.go]: Add AMIS Header

File: cmd/istanbul/cmd.go (hunk 1 above)
Comment (bailantaotao) [inline cmd/istanbul/cmd.go]: OK

File: cmd/istanbul/cmd.go [hunk 8]
```diff
@@ -0,0 +1,127 @@
+package main
```
Comment (markya0616) [inline cmd/istanbul/cmd.go]: Add AMIS Header

File: README.md (hunk 2 above)
Comment (bailantaotao) [inline README.md]: OK

File: cmd/utils/app.go [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package utils
```
Comment (markya0616) [inline cmd/utils/app.go]: Add AMIS Header

File: cmd/istanbul/flags.go (hunk 7 above)
Comment (bailantaotao) [inline cmd/istanbul/flags.go]: OK

File: cmd/istanbul/cmd.go (hunk 8 above)
Comment (bailantaotao) [inline cmd/istanbul/cmd.go]: OK

File: cmd/utils/app.go (hunk 9 above)
Comment (bailantaotao) [inline cmd/utils/app.go]: OK

File: cmd/istanbul/cmd.go (hunk 6 above)
Comment (bailantaotao) [inline cmd/istanbul/cmd.go]: Reference identifier. 
This will be removed when Istanbul PR is merged

File: cmd/istanbul/cmd.go (hunk 5 above)
Comment (bailantaotao) [inline cmd/istanbul/cmd.go]: OK

File: cmd/istanbul/cmd.go (hunk 4 above)
Comment (bailantaotao) [inline cmd/istanbul/cmd.go]: OK

File: cmd/istanbul/main.go (hunk 3 above)
Comment (bailantaotao) [inline cmd/istanbul/main.go]: OK

**Pull request merged**